src/Payum/Server/Action/AuthorizePaymentAction.php

```diff
@@ -1,24 +1,29 @@
 <?php
+declare(strict_types=1);
+
 namespace Payum\Server\Action;
 
-use Payum\Core\Action\GatewayAwareAction;
+use Payum\Core\Action\ActionInterface;
 use Payum\Core\Bridge\Spl\ArrayObject;
 use Payum\Core\Exception\RequestNotSupportedException;
+use Payum\Core\GatewayAwareTrait;
 use Payum\Core\Model\Payment as PayumPayment;
 use Payum\Core\Request\Authorize;
 use Payum\Core\Request\Convert;
 use Payum\Core\Request\GetHumanStatus;
 use Payum\Server\Model\Payment;
 use Payum\Server\Request\ObtainMissingDetailsRequest;
 
-class AuthorizePaymentAction extends GatewayAwareAction
+class AuthorizePaymentAction implements ActionInterface
```

Comment:
should also implement GatewayAwareInterface

Reply:
GatewayAwareTrait is not enough

Reply:
Added. Thanks!